src/awkward/_v2/contents/listoffsetarray.py

```diff
@@ -147,6 +147,95 @@ def _carry(self, carry, allow_lazy, exception):
             self._parameters,
         )
 
+    def compact_offsets64(self, start_at_zero):
+        nplike = self.nplike
+        out = ak._v2.index.Index64.empty(self._length, self.nplike)
+        self._handle_error(
+            nplike[
+                "awkward_ListOffsetArray_compact_offsets",
+                out.dtype.type,
+                self._offsets.dtype.type,
+            ](
+                out.to(nplike),
+                self._offsets.to(nplike),
+                self._length,
+            )
+        )
+        return out
+
+    def broadcast_tooffsets64(self, offsets):
```

Comment:
If `compact_offsets64` and `broadcast_tooffset64` are very similar between ListArray and ListOffsetArray, we should find some way to combine them.

For example, the `minmax_depth` method on Form was repurposed in Content using a sneaky Python trick (the implementation duck-types to both classes, so it can just be referenced). This is a scale on which it will be beneficial to make the code DRY. (Sometimes people work too hard to make very short segments DRY, but it really matters on these 100 line functions.)